### PR TITLE
[FLINK-4255] Unstable test WebRuntimeMonitorITCase.testNoEscape

### PR DESCRIPTION
**Pull request by BorisOsipov**
Fix two issues:
[FLINK-4255] Unstable test WebRuntimeMonitorITCase.testNoEscape
[FLINK-3746] WebRuntimeMonitorITCase.testNoCopyFromJar failing intermittently
- wait until JobManager will be available
- code simplification

### CODE REVIEW

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java [hunk 1]
```diff
@@ -491,6 +445,37 @@ public void testNoCopyFromJar() throws Exception {
...
+		WebRuntimeMonitor webMonitor = new WebRuntimeMonitor(
+			config,
+			flink.createLeaderRetrievalService(),
+			jmActorSystem);
+
+		webMonitor.start(jobManagerAddress);
+		JobManagerRetriever retriever = Whitebox
+            .getInternalState(webMonitor, "retriever");
+
+		retriever.awaitJobManagerGatewayAndWebPort();
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: instead you can call `flink.waitForActorsToBeAlive()`

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java [hunk 2]
```diff
@@ -75,32 +78,14 @@
 	public void testStandaloneWebRuntimeMonitor() throws Exception {
 		final Deadline deadline = TestTimeout.fromNow();
 
-		TestingCluster flink = null;
+		TestingCluster testingCluster = null;
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: please undo the renaming

File: flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java [hunk 3]
```diff
@@ -262,8 +247,8 @@ public void testRedirectToLeader() throws Exception {
 				response = followingClient.getNextResponse(deadline.timeLeft());
 				assertEquals(HttpResponseStatus.OK, response.getStatus());
 				assertEquals(response.getType(), MimeTypes.getMimeTypeForExtension("json"));
-				assertTrue(response.getContent().contains("\"taskmanagers\":1") ||
-						response.getContent().contains("\"taskmanagers\":0"));
+				assertThat(response.getContent(), either(containsString("\"taskmanagers\":1"))
```
Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: Why did you change this line? (Note that this applies to all lines that replace assertTrue with assertThat, contains with containsString and | with or.)

Comment (BorisOsipov) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: I've replaced assertTrue/False because assertThat and hamcrest matchers provide much better error messages. Instead of weird 'java.lang.AssertionError' we can get a more readable message in the log without effort.
Of cource that's a matter of opinion and I can revert it. Should I do?

Comment (zentol) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: I see. The thing is that the entire codebase uses the junit asserts, and i would rather not break this consistency unless for a very good reason, and this seems to minor imo.

It was a good idea though ;)

Comment (BorisOsipov) [inline flink-runtime-web/src/test/java/org/apache/flink/runtime/webmonitor/WebRuntimeMonitorITCase.java]: Ok. Let's save consistency.

**Comment by BorisOsipov** [comment]
@zentol Thank you for the review. I've resolved your comments.

**Comment by zentol** [comment]
Merging.
